### PR TITLE
#56 fix username already exist exception and invalid login exception

### PR DESCRIPTION
**Pull request by Joexiaohongchen**

### CODE REVIEW

File: backend/build.gradle [hunk 1]
```diff
@@ -17,13 +17,16 @@ dependencies {
...
+    implementation 'org.springframework.boot:spring-boot-starter-validation'
     implementation 'org.projectlombok:lombok:1.18.12'
     implementation 'org.springframework.boot:spring-boot-starter-jdbc:2.3.2.RELEASE'
+    compile group: 'javax.persistence', name: 'javax.persistence-api', version: '2.2'
     implementation 'io.jsonwebtoken:jjwt:0.9.0'
     implementation 'com.alibaba:fastjson:1.2.36'
     runtimeOnly 'mysql:mysql-connector-java:8.0.21'
     runtimeOnly 'org.springframework.boot:spring-boot-devtools:2.3.2.RELEASE'
     testImplementation 'org.springframework.boot:spring-boot-starter-test:2.3.2.RELEASE'
+
```
Comment (hxue98) [inline backend/build.gradle]: Remove empty line

File: backend/src/main/java/team/YongAndJoe/NewsTodayBackend/controller/AccountController.java [hunk 2]
```diff
@@ -1,15 +1,18 @@
 package team.YongAndJoe.NewsTodayBackend.controller;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
```
Comment (hxue98) [inline backend/src/main/java/team/YongAndJoe/NewsTodayBackend/controller/AccountController.java]: Do not use * in import.

File: backend/src/main/java/team/YongAndJoe/NewsTodayBackend/entity/User.java [hunk 3]
```diff
@@ -29,6 +37,14 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    public String getConfirmpassword() {
```
Comment (hxue98) [inline backend/src/main/java/team/YongAndJoe/NewsTodayBackend/entity/User.java]: Do not need confirm password in backend. Only frontend needs to confirm password before making the request.

File: backend/src/main/java/team/YongAndJoe/NewsTodayBackend/util/AjaxResponseBody.java [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
+import team.YongAndJoe.NewsTodayBackend.entity.User;
+
 import java.io.Serializable;
 
 public class AjaxResponseBody implements Serializable {
     private boolean success;
     private String msg;
     private Object result;
     private String jwtToken;
+    private String username;
```
Comment (hxue98) [inline backend/src/main/java/team/YongAndJoe/NewsTodayBackend/util/AjaxResponseBody.java]: Remove username

File: backend/src/main/java/team/YongAndJoe/NewsTodayBackend/validator/LoginValidator.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    AccountService accountService;
+
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return User.class.equals(clazz);
+    }
+
+    @Override
+    public void validate(Object object, Errors errors) {
+
```
Comment (hxue98) [inline backend/src/main/java/team/YongAndJoe/NewsTodayBackend/validator/LoginValidator.java]: Remove empty line

File: backend/src/main/java/team/YongAndJoe/NewsTodayBackend/validator/LoginValidator.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        User user = (User) object;
+        if(user.getUsername().length() == 0) {
+            errors.rejectValue("username", "Length", "Username is required");
+        } else if(!accountService.existByUsername(user.getUsername())){
```
Comment (hxue98) [inline backend/src/main/java/team/YongAndJoe/NewsTodayBackend/validator/LoginValidator.java:30]: Add check for password length

File: database/.vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
```
Comment (hxue98) [inline database/.vscode/settings.json]: add .gitignore to database and frontend folder.

**Review by hxue98 — CHANGES_REQUESTED** [review]

File: backend/src/main/java/team/YongAndJoe/NewsTodayBackend/controller/AccountController.java [hunk 8]
```diff
@@ -23,38 +26,48 @@
...
+    public ResponseEntity<?> login(@RequestBody User user, BindingResult result) {
+
+        loginValidator.validate(user, result);
+        ResponseEntity<?> errorMap = mapValidationErrorService.mapValidationService(result);
+        if(errorMap != null) return errorMap;
+
         AjaxResponseBody body = new AjaxResponseBody();
+        body.setSuccess(true);
+        body.setUsername(user.getUsername());
+//        body.setJwtToken(jwtTokenUtil.generateJwt(user));
```
Comment (hxue98) [inline backend/src/main/java/team/YongAndJoe/NewsTodayBackend/controller/AccountController.java]: Uncomment this line

**Review by hxue98 — CHANGES_REQUESTED** [review]

**Comment by Joexiaohongchen** [comment]
All updated, please review.

**Review by hxue98 — APPROVED** [review]

**Pull request merged**